### DISCUSSION TITLE
How to debug app after installed via TestFlight

### DISCUSSION THREAD

**Discussion by SenexTech** [dotnet/maui#8790] [Q&A]
What is the recommended method to debug apps that crash emmediatly on opening after being deployed through Apple Store and TestFlight.   I have an app that runs fine when debuging to a local iOS device (iPhone) but when I distribut the same app to testers through Apple store and TestFlight it crashes emmediatly when starting.  I thought of using App Center to get crash reports but that is not compatible with MAUI.   BTW I do not have a physical Mac but use a hosted Mac in the cloud. My only alternative now is to strip my app back to the basic template and start adding back one item at a time and then uploading to Apple one change at a time until it crashes.  (Basic MAUI template does not crash) . This is very time consuimg and there has to be a better way.

**Comment by sencagri** [comment]
There is no debugging on physical device at the moment as I know.

**Comment by davidbuckleyni** [comment]
U need use something like app centre or ray gun and log ur app heavly nothings to stop u sending ur self a log file from app

**Comment by JohnHDev** [comment]
Im surprised AppCenter crash reporting doesn't work, but you could wrap the startup code in a try except, store as much detail as you can in a local file (crash.txt for ex), then on app startup check for that file before you load your main page. If it exists, go to a VERY basic page asking to email that file (and also delete the file so the user isn't stuck in a loop when you manage to fix the issue).

I would expect with a few builds you should be able to diagnose the issue, it shouldn't take too long, if you haven't already managed it.
